docs: add doc about custom resources and parameters (#3255)
aws-amplify/docs: docs/cli/graphql-transformer/resolvers.md

## Custom Resolvers

> GraphQL transformer supports adding custom `Query`, `Mutation` and `Subscription` when the generated ones do not cover your use-case. This is achieved by adding the required `Query`, `Mutation` or `Subscription` type to your schema and then adding resolvers to newly added field by creating a custom stack inside `<project-root>/amplify/backend/api/<api-name>/stacks` directory of your API. To add the custom fields add the following to your schema

Reviewer: ```suggestion
You can add custom `Query`, `Mutation` and `Subscription` when the generated ones do not cover your use case. To do this, first add the required `Query`, `Mutation` or `Subscription` type to your schema. Then, add resolvers for the newly added fields by creating a custom stack inside `<project-root>/amplify/backend/api/<api-name>/stacks` directory of your API.

To add the custom fields, add the following to your schema:
```

```graphql
  # <project-root>amplify/backend/api/<api-name>/schema.graphql

  type Query {
    # Add all the custom queries here
  }

  type Mutation {
    # Add all the custom mutations here
  }

  type Subscription {
    # Add all the custom subscription here
  }

```

> Transformer by default creates a file called `CustomResources.json` inside `<project-root>/amplify/backend/api/<api-name>/stacks`, which can be used to add the custom resolvers for newly added `Query`, `Mutation` or `Subscription`. The custom stack gets the following arguments passed to it, allowing you to get details about API

Reviewer: ```suggestion
The GraphQL Transformer by default creates a file called `CustomResources.json` inside `<project-root>/amplify/backend/api/<api-name>/stacks`, which can be used to add the custom resolvers for newly added `Query`, `Mutation` or `Subscription`. The custom stack gets the following arguments passed to it, allowing you to get details about API:
```

                    "Fn::Sub": [
                        "s3://${S3DeploymentBucket}/${S3DeploymentRootKey}/resolvers/Query.myCustomQuery.req.vtl",
                        {
                            "S3DeploymentBucket": {
                                "Ref": "S3DeploymentBucket"
                            },
                            "S3DeploymentRootKey": {
                                "Ref": "S3DeploymentRootKey"
                            }
                        }
                    ]
                },
                "ResponseMappingTemplateS3Location": {
                    "Fn::Sub": [
                        "s3://${S3DeploymentBucket}/${S3DeploymentRootKey}/resolvers/Query.myCustomQuery.res.vtl",
                        {
                            "S3DeploymentBucket": {
                                "Ref": "S3DeploymentBucket"
                            },
                            "S3DeploymentRootKey": {
                                "Ref": "S3DeploymentRootKey"
                            }
                        }
                    ]
                }
            }
        }
    }
}
```

> The request and response template should be placed inside `<project-root>/amplify/backend/api/<api-name>/resolvers` folder. Resolver templates are written using VTL. For more details refer to [AWS AppSync VTL documentation](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-mapping-template-reference.html)

Reviewer: what about the resolver file name? We should add a note on its format.

Author: When these are custom the customer can name it however they see it fit. They should use the same name in the `CustomResources.json`

Reviewer: I'd recommend putting a default here that we recommend.

## Custom Resolvers

> You can add custom `Query`, `Mutation` and `Subscription` when the generated ones do not cover your use case. To do this, first add the required `Query`, `Mutation` or `Subscription` type to your schema. Then, add resolvers for the newly added fields by creating a custom stack inside `<project-root>/amplify/backend/api/<api-name>/stacks` directory of your API.

Reviewer: How about this:
```suggestion
You can add custom `Query`, `Mutation` and `Subscription` when the generated ones do not cover your use case. 

1. Add the required `Query`, `Mutation` or `Subscription` type to your schema. 
2. Add resolvers for the newly added fields by creating a custom stack inside `<project-root>/amplify/backend/api/<api-name>/stacks` directory of your API
3. Edit the resolver ETC ETC ETC ETC
```
